### PR TITLE
Add new methods

### PR DESCRIPTION
**Pull request by inga-paladi**

### CODE REVIEW

File: MedTracker/Controllers/DoctorController.cs [hunk 1]
```diff
@@ -19,7 +19,6 @@ public DoctorController(ApplicationDbContext context)
     {
         _context = context;
     }
-
```
Comment (inga-paladi) [inline MedTracker/Controllers/DoctorController.cs]: unnecessary change

File: MedTracker/Controllers/PatientController.cs [hunk 2]
```diff
@@ -117,6 +117,31 @@ public async Task<IActionResult> GetTreatments()
...
+    [HttpGet("GetDoctors")]
+    public async Task<IActionResult> GetDoctors()
+    {
+        try
+        {
+            // Fetch doctors and their information
+            var doctors = await _context.Doctors
+                .Select(d => new
+                {   
+                    DocID = d.IdUser,
```
Comment (inga-paladi) [inline MedTracker/Controllers/PatientController.cs]: Please, use normal names.
Why first name is FName and phone number is PhoneNumber. If the intention was to shorten the name, it is not applied to all of them.
The names should be: DoctorId, FirstName, LastName, etc

File: MedTracker/Controllers/DoctorController.cs [hunk 3]
```diff
@@ -110,6 +109,83 @@ public async Task<IActionResult> GetTreatments()
             return StatusCode(500, $"An error occurred:{ex.Message}");
         }
     }
+    [HttpGet("GetPatient")]
+    [Authorize]
+    public async Task<IActionResult> GetPateint()
```
Comment (inga-paladi) [inline MedTracker/Controllers/DoctorController.cs]: GetPat**eint**

File: MedTracker/Controllers/DoctorController.cs [hunk 4]
```diff
@@ -110,6 +109,83 @@ public async Task<IActionResult> GetTreatments()
             return StatusCode(500, $"An error occurred:{ex.Message}");
         }
     }
```
Comment (inga-paladi) [inline MedTracker/Controllers/DoctorController.cs:112]: It would be a good idea to add a new line to separate the methods

File: MedTracker/Controllers/DoctorController.cs (hunk 3 above)
Comment (inga-paladi) [inline MedTracker/Controllers/DoctorController.cs]: Lets look just at the code, not at the task you should do.
We see method `GetPatient`. We understand this is singular form and an id should be given as argument, but I don't see one.

This method returns a list of `treatmentsWithPatients`, here becomes even more confusing. Are we receiving a patient, a list of patients, a list of treatments or a list of treatments with patients?
Based on the code we can't tell what are the intentions.

File: MedTracker/Controllers/PatientController.cs [hunk 5]
```diff
@@ -117,6 +117,31 @@ public async Task<IActionResult> GetTreatments()
 
     }
 
+    [HttpGet("GetDoctors")]
+    public async Task<IActionResult> GetDoctors()
```
Comment (inga-paladi) [inline MedTracker/Controllers/PatientController.cs:122]: If we don't use any input data, at least we should require the user to authenticate itself.

File: MedTracker/Controllers/DoctorController.cs (hunk 3 above)
Comment (inga-paladi) [reply MedTracker/Controllers/DoctorController.cs]: Still valid

Comment (ninico11) [reply MedTracker/Controllers/DoctorController.cs]: Now it is GetPatients and it return a list of treatments with patients. In example wroted by you upper is method is GetTreatments and it return the list of treatments with medications. How would you like to name this method?

Comment (inga-paladi) [reply MedTracker/Controllers/DoctorController.cs]: > Now it is GetPatients and it return a list of treatments with patients

If it was at least `patients WITH treatments`, but you are returning `treatments WITH patients`.

> In example wroted by you upper is method is GetTreatments

Yes, there is already a method that returns treatments. Why to make another one?

> In example wroted by you upper is method is GetTreatments and it return the list of treatments with medications

A patient without a treatment is not a patient anymore, is simply a "healthy" human. But what is a treatment without medications?
Ideally yes, if they are treated as two entities, there could have been two methods, one for treatment, as it is in database, and another just for medications. Thanks for pointing me out on this, I made that at night in hurry :) I think I'll split that in another PR.

> How would you like to name this method?

Not renaming, but returning just the patients.

Comment (ninico11) [reply MedTracker/Controllers/DoctorController.cs]: I will try to return patients with treatments because in another way how would we know which treatments have which patients

Comment (ninico11) [reply MedTracker/Controllers/DoctorController.cs]: Now it returns patients with treatments

Comment (inga-paladi) [reply MedTracker/Controllers/DoctorController.cs]: > Now it returns patients with treatments

But I asked to return just the patients in this method.

> how would we know which treatments have which patients

By creating another method "GetPatientTreatments", which will receive the PatientId.

Comment (ninico11) [reply MedTracker/Controllers/DoctorController.cs]: I added new method GetPatientTreatments which return id of a patient and treatment and modified method GetPatient to return information about patient

File: MedTracker/Controllers/DoctorController.cs [hunk 6]
```diff
@@ -139,7 +139,6 @@ public async Task<IActionResult> GetPatients()
                 .Select(t => new
```
Comment (inga-paladi) [inline MedTracker/Controllers/DoctorController.cs:139]: You can combine this and the query bellow using `Join`s.

File: MedTracker/Controllers/DoctorController.cs [hunk 7]
```diff
@@ -154,33 +153,68 @@ public async Task<IActionResult> GetPatients()
...
+            return Ok(patients);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"An error occurred: {ex.Message}");
+        }
+    }
 
-            var patientsWithTreatments = new List<object>();
+    [HttpGet("GetPatientTreatments")]
```
Comment (inga-paladi) [inline MedTracker/Controllers/DoctorController.cs:171]: We need to receive the id for the user whose treatments we are interested. Just look at the name: is `GetPatient_Treatments`, not `GetTreatments`.  There are some methods which receive just one id as input, use them as reference for this method.

Comment (ninico11) [reply MedTracker/Controllers/DoctorController.cs:171]: We need receive all patients and all their treatments for the list of treatments in front. And after this if we want to see another patient info i can do another method but menthors dont mentionated it. We need only name and phone number.

File: MedTracker/Controllers/DoctorController.cs (hunk 6 above)
Comment (ninico11) [reply MedTracker/Controllers/DoctorController.cs:139]: I dont understand this problem

Comment (inga-paladi) [reply MedTracker/Controllers/DoctorController.cs:139]: Clean code perspective.
Just look at the first query. You are selecting `IdTreatment` and `DoctorID`, but you are not even using `DoctorID` further in code. The line 146 could be removed, if you selected from the start just `ItTreatment`. Second, the code becomes smaller with just one query. More lines of code, means more lines to read and to understand (in our case, not every time)

File: MedTracker/Controllers/DoctorController.cs (hunk 7 above)
Comment (inga-paladi) [reply MedTracker/Controllers/DoctorController.cs:171]: this is not ok from api perspective. you mean we should return bulk data every time we need some data? To have method GetPatients(), GetPatientsWithTreatments(), GetPatientsAndTeirDoctors(), GetTreatmentsAndDoctorsDataAndPatientData()?

Comment (inga-paladi) [reply MedTracker/Controllers/DoctorController.cs:171]: But ok, **just** for the demo, I'll merge this

File: MedTracker/Controllers/DoctorController.cs (hunk 6 above)
Comment (ninico11) [reply MedTracker/Controllers/DoctorController.cs:139]: Now it is a single query
